### PR TITLE
add test api transformer_int4_fp16_gpu

### PR DESCRIPTION
**Pull request by pengyb2001**
## Description
Add test api transformer_int4_fp16_gpu in all-in-one benchmark. The attached files are results of two apis.
[transformer_int4_fp16_gpu-results-2024-04-02.csv](https://github.com/intel-analytics/ipex-llm/files/14834323/transformer_int4_fp16_gpu-results-2024-04-02.csv)
[transformer_int4_gpu-results-2024-04-02.csv](https://github.com/intel-analytics/ipex-llm/files/14834325/transformer_int4_gpu-results-2024-04-02.csv)

### CODE REVIEW

File: python/llm/dev/benchmark/all-in-one/run.py [hunk 1]
```diff
@@ -427,6 +430,11 @@ def run_transformer_int4_gpu(repo_id,
                                                             trust_remote_code=True, use_cache=True).eval()
         tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
         model = model.to('xpu')
+
+    if fp16:
+        model = model.half()
+        print("Convert model to half precision")
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/run.py:436]: better first do half and then put to xpu to save xpu memory?

File: python/llm/dev/benchmark/all-in-one/config.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 repo_id:
-  - 'THUDM/chatglm-6b'
-  - 'THUDM/chatglm2-6b'
+  # - 'THUDM/chatglm-6b'
+  # - 'THUDM/chatglm2-6b'
   - 'meta-llama/Llama-2-7b-chat-hf'
   # - 'liuhaotian/llava-v1.5-7b' # requires a LLAVA_REPO_DIR env variables pointing to the llava dir; added only for gpu win related test_api now
-local_model_hub: 'path to your local model hub'
+local_model_hub: '/mnt/disk1/models'
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/config.yaml:6]: revert this

**Comment by pengyb2001** [comment]
updated @hkvision

File: python/llm/dev/benchmark/all-in-one/README.md [hunk 3]
```diff
@@ -37,22 +36,30 @@ in_out_pairs:
   - '32-32'
   - '1024-128'
 test_api:
-  - "transformer_int4"
-  - "native_int4"
-  - "optimize_model"
-  - "pytorch_autocast_bf16"
-  # - "transformer_autocast_bf16"
   # - "ipex_fp16_gpu" # on Intel GPU
   # - "bigdl_fp16_gpu" # on Intel GPU
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/README.md:40]: put them after `transformer_int4_fp16_gpu`?

Comment (pengyb2001) [reply python/llm/dev/benchmark/all-in-one/README.md]: modified

File: python/llm/dev/benchmark/all-in-one/README.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 All in one benchmark test allows users to test all the benchmarks and record them in a result CSV. Users can provide models and related information in `config.yaml`.
 
-Before running, make sure to have [bigdl-llm](../../../README.md).
+Before running, make sure to have [ipex-llm](../../../../../README.md).
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/README.md:5]: have ipex-llm installed

File: python/llm/dev/benchmark/all-in-one/README.md [hunk 5]
```diff
@@ -37,22 +36,30 @@ in_out_pairs:
...
-  - "optimize_model"
-  - "pytorch_autocast_bf16"
-  # - "transformer_autocast_bf16"
+  - "transformer_int4_gpu"  # on Intel GPU
+  # - "transformer_int4_fp16_gpu" # on Intel GPU, use fp16 for non-linear layer
   # - "ipex_fp16_gpu" # on Intel GPU
   # - "bigdl_fp16_gpu" # on Intel GPU
-  # - "transformer_int4_gpu"  # on Intel GPU
   # - "optimize_model_gpu"  # on Intel GPU
   # - "deepspeed_transformer_int4_cpu" # on Intel SPR Server
```
Comment (hkvision) [inline python/llm/dev/benchmark/all-in-one/README.md:44]: put this cpu together with other cpu after gpu?

**Review by hkvision — APPROVED** [review]

**Pull request merged**